packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -77,25 +104,181 @@ function WorklistDataGrid(props) {
+    onRowSelectAll,
+    onClearSelectedRows,
+    onDisableSelectableRows,
+    hasSelectableRows,
     intl,
+    rowHeaderIndex,
   } = props;
 
+  const grid = useRef();
   const focusedRow = useRef(0);
   const focusedCol = useRef(0);
-  const grid = useRef();
+  const ariaLiveMsg = useRef();
+
+  const [currentSelectedCell, setCurrentSelectedCell] = useState(null);
+  const [, setRowSelectionModeMessage] = useState();
+  const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(columns);
+
+  const theme = useContext(ThemeContext);
+
+  const isRowSelectionCell = (columnIndex) => (
+    hasSelectableRows && columnIndex < displayedColumns.length && displayedColumns[columnIndex].id === WorklistDataGridUtils.ROW_SELECTION_COLUMN.id
+  );
 
-  const gridRef = useCallback((node) => {
-    grid.current = node;
-    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+  const isCellSelected = (rowId, columnId) => (currentSelectedCell && currentSelectedCell.rowId === rowId && currentSelectedCell.columnId === columnId);
+
+  const setFocus = (rowIndex, colIndex, makeActiveElement) => {
+    let focusedCell = grid.current.rows[rowIndex].cells[colIndex];
+    if (isRowSelectionCell(colIndex) && focusedCell.firstChild) {
+      focusedCell = focusedCell.firstChild;
+    }
     focusedCell.tabIndex = 0;
-  }, []);
+    if (makeActiveElement && focusedCell.focus) {
+      focusedCell.focus();
+    }
+  };
 
-  const handleKeyDown = (event) => {
-    const currentFocusedRow = event.target.parentElement.rowIndex;
-    const currentFocusedCol = event.target.cellIndex;
+  const removeTabStop = (rowIndex, colIndex) => {
+    const cell = grid.current.rows[rowIndex].cells[colIndex];
+    // Remove Tab stop from previous cell in table that has focus and set it to the
+    // cell that was clicked.
+    cell.tabIndex = -1;
+    if (isRowSelectionCell(colIndex) && cell.getElementsByTagName('input').length > 0) {
+      // For row selection cell, the tabstop won't be on the cell itself but the
+      // inner input element so remove it from the input element.
+      cell.getElementsByTagName('input')[0].tabIndex = -1;
+    }
+  };
+
+  const setFocusedRowCol = (newRowIndex, newColIndex, makeActiveElement) => {
+    removeTabStop(focusedRow.current, focusedCol.current);
+    focusedRow.current = newRowIndex;
+    focusedCol.current = newColIndex;
+    setFocus(newRowIndex, newColIndex, makeActiveElement);
+  };
+
+  useEffect(() => {
+    // When row selection mode is turned on or off a row selection column is added or removed.
+    // Therefore, shift the focused cell to the left or right.
+    let newFocusCell = { row: focusedRow.current, col: (focusedCol.current + (hasSelectableRows ? 1 : -1)) };
+
+    if (hasSelectableRows && focusedRow.current === 0 && focusedCol.current === 0) {
+      // When row selection is turned on for the first time, default to the first row selection cell.
```

Comment:
This does not seem correct.  Should we be taking focus away from our parent component on our own?  Just because we have been rendered does not mean we should take focus.

Reply:
It does not take focus from the parent but we need to know where focus should go when they tab into the grid.